modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -375,6 +375,32 @@ impl LightningClientModule {
         Ok(operation_id)
     }
 
+    async fn get_operation_id(&self, invoice: &Bolt11Invoice) -> anyhow::Result<OperationId> {
+        for payment_attempt in 0..u64::MAX {
+            let operation_id = OperationId::from_encodable((invoice.clone(), payment_attempt));
+
+            if self.client_ctx.has_active_states(operation_id).await {
+                bail!("Previous payment attempt still in progress. Previous Operation Id: {operation_id}");
+            } else if self.client_ctx.has_inactive_states(operation_id).await {
+                let mut stream = self
+                    .subscribe_send(operation_id)
+                    .await
+                    .expect("operation_id has inactive states")
+                    .into_stream();
+
+                while let Ok(state) = stream.ok().await {
```

Comment:
If all you do is matching `Ok(_)` then you could just use `next` here and match `Some(_)`.